### PR TITLE
[test] Add test to resolve node with russian idtf

### LINKED ISSUE #127: Generation of the node with Russian identifier
**Describe the bug**
After resolving a node with a Russian identifier search return empty addr of node

**To Reproduce**
1. Generate node with the identifier on Russian
2. Find node by the same identifier

**Expected behavior**
Node should be returned, or shouldn't be generated 

**Screenshots**
image

**Desktop (please complete the following information):**
 - Version [e.g. 0.7.0]

### PR DESCRIPTION
**Pull request by smousy**
* [x] Read PR [documentation](https://github.com/ostis-ai/sc-machine/blob/main/CONTRIBUTING.md)
* [x] Update changelog
* [x] Update documentation
Fixed #127

### CODE REVIEW

**Review by KovalM — CHANGES_REQUESTED** [review]
We should forbid to use Cyrillic for system idtf

File: sc-memory/sc-core/sc_helper.c [hunk 1]
```diff
@@ -223,6 +231,14 @@ sc_result sc_helper_set_system_identifier(sc_memory_context * ctx, sc_addr addr,
 
   sc_assert(sc_keynodes != null_ptr);
 
+  regex_t regex;
+  regcomp(&regex, "([a-z]|[A-Z]|'_'|'.'|[0-9])+", REG_EXTENDED);
```
Comment (KovalM) [inline sc-memory/sc-core/sc_helper.c:235]: Move regex to constants. And move similar code to function or macros

**Review by KovalM — CHANGES_REQUESTED** [review]

**Review by KovalM — APPROVED** [review]

**Review by NikitaZotov — COMMENTED** [review]
Add changelog message

File: sc-memory/sc-core/sc_helper.c [hunk 2]
```diff
@@ -154,6 +155,18 @@ void sc_helper_shutdown()
   _destroy_keynodes_str();
 }
 
+sc_result sc_helper_check_system_identifier(const sc_char * data)
```
Comment (NikitaZotov) [inline sc-memory/sc-core/sc_helper.c:158]: Please, add function definition to header `sc-helper.h`

**Review by NikitaZotov — APPROVED** [review]

**Pull request merged**